app/controllers/api/v1/subjects_controller.rb

```diff
@@ -98,6 +98,8 @@ def context
   def location_params(locations)
     (locations || []).map.with_index do |loc, i|
       location_params = case loc
+                        when *Subject.nonstandard_mimetypes.keys
```

Comment:
Not actually, The Hound. Because this matches both the splat array check and the type check, the type isn't replaced with that `when` at the bottom. I didn't want to bust up this classy block.

Reply:
Could this new when just collapse to the existing string one where it does the lookup?
```
when String
  { content_type: Subject.nonstandard_mimetypes[loc] || loc }
```